pkg/lenses/stepgraph/stepgraph.go

```diff
@@ -0,0 +1,105 @@
+package stepgraph
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"sort"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/spyglass/api"
+	"k8s.io/test-infra/prow/spyglass/lenses"
+	"sigs.k8s.io/yaml"
+
+	citoolsapi "github.com/openshift/ci-tools/pkg/api"
+)
+
+const (
+	name     = "steps"
+	title    = "CI-Operator steps"
+	priority = 6
+)
+
+// Lens is the implementation of a JUnit-rendering Spyglass lens.
+type Lens struct{}
+
+// Config returns the lens's configuration.
+func (lens Lens) Config() lenses.LensConfig {
+	return lenses.LensConfig{
+		Name:     name,
+		Title:    title,
+		Priority: priority,
+	}
+}
+
+var tmpl *template.Template
+
+func init() {
+	data, err := staticTemplateHtmlBytes()
+	if err != nil {
+		panic(fmt.Sprintf("failed to gzunip template: %v", err))
+	}
+	tmpl = template.Must(template.New("template").Parse(string(data)))
+}
+
+// Header renders the content of <head> from template.html.
+func (lens Lens) Header(artifacts []api.Artifact, _ string, config json.RawMessage) string {
+	var buf bytes.Buffer
+	if err := tmpl.ExecuteTemplate(&buf, "header", nil); err != nil {
+		return fmt.Sprintf("<!-- FAILED EXECUTING HEADER TEMPLATE: %v -->", err)
+	}
+	return buf.String()
+}
+
+// Callback does nothing.
```

Comment:
Explain why? This is to handle click-throughs inside of lenses, right?

Reply:
This is to handle fetching additional data after the page was already loaded to not require all data during initial load. The only upstream lens where this does something is the buildlog lens, but it's part of the interface so it needs to exist.